d2common/d2fileformats/d2mpq/mpq_block.go

```diff
@@ -0,0 +1,77 @@
+package d2mpq
+
+import (
+	"io"
+	"strings"
+)
+
+// FileFlag represents flags for a file record in the MPQ archive
+type FileFlag uint32
+
+const (
+	// FileImplode - File is compressed using PKWARE Data compression library
+	FileImplode FileFlag = 0x00000100
+	// FileCompress - File is compressed using combination of compression methods
+	FileCompress FileFlag = 0x00000200
+	// FileEncrypted - The file is encrypted
+	FileEncrypted FileFlag = 0x00010000
+	// FileFixKey - The decryption key for the file is altered according to the position of the file in the archive
+	FileFixKey FileFlag = 0x00020000
+	// FilePatchFile - The file contains incremental patch for an existing file in base MPQ
+	FilePatchFile FileFlag = 0x00100000
+	// FileSingleUnit - Instead of being divided to 0x1000-bytes blocks, the file is stored as single unit
+	FileSingleUnit FileFlag = 0x01000000
+	// FileDeleteMarker - File is a deletion marker, indicating that the file no longer exists. This is used to allow patch
+	// archives to delete files present in lower-priority archives in the search chain. The file usually
+	// has length of 0 or 1 byte and its name is a hash
+	FileDeleteMarker FileFlag = 0x02000000
+	// FileSectorCrc - File has checksums for each sector. Ignored if file is not compressed or imploded.
+	FileSectorCrc FileFlag = 0x04000000
+	// FileExists - Set if file exists, reset when the file was deleted
+	FileExists FileFlag = 0x80000000
+)
```

Comment:
These flags are also referenced in `mpq_stream.go`, maybe they should go into their own `file_flags.go` file.